### PR TITLE
[SUBS-1377] Refactored the error handling

### PR DESCRIPTION
**Pull request by ke4**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java [hunk 1]
```diff
@@ -31,7 +32,7 @@
     private static final String ERROR_RESULT_BEGINNING = "ERROR:";
     private static final String OK_RESULT_BEGINNING = "OK";
 
-    private String validationError;
+    private List<String> validationErrors;
```
Comment (fpenim) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java]: Why do we need to have the validation errors list as a class variable?

Comment (fpenim) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java]: Why do we need to have the validation errors list as a class variable?

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java [hunk 2]
```diff
@@ -50,11 +53,19 @@ public SingleValidationResult validateFileContent() throws IOException, Interrup
         if (last >= 0) {
             resultMessage = output.substring(last, output.length());
             if (resultMessage.contains(ERROR_RESULT_BEGINNING)) {
-                validationError = resultMessage.replace(ERROR_RESULT_BEGINNING, "").trim();
+                validationErrors.add(resultMessage.replace(ERROR_RESULT_BEGINNING, "").trim());
```
Comment (fpenim) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java]: This should return a singleValidationError object with the error message you build here.

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java [hunk 3]
```diff
@@ -84,31 +95,47 @@ private String assembleValidatorCommand() {
...
+
         validateFileExistence();
 
         validateFileType();
 
-        return true;
+        return validationErrors.size() == 0;
+    }
+
+    private void initValidationErrors() {
```
Comment (fpenim) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java]: Unnecessary if the list is not defined as a variable of the class.

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java [hunk 4]
```diff
@@ -84,31 +95,47 @@ private String assembleValidatorCommand() {
...
+        this.validationErrors = new ArrayList<>();
     }
 
     private void validateFileExistence() {
-        File file = new File(getCommandLineParams().getFilePath());
+        String filePath = getCommandLineParams().getFilePath();
+        File file = new File(filePath);
         if(!file.exists() || file.isDirectory()) {
-            throw new FileNotFoundException(getCommandLineParams().getFilePath());
+            this.validationErrors.add(String.format(ErrorMessages.FILE_NOT_FOUND_BY_TARGET_PATH, filePath));
```
Comment (fpenim) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java]: As above, this should return a SingleValidationResult object

**Review by fpenim — CHANGES_REQUESTED** [review]
A few things need to be clarified/changed

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java (hunk 4 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java]: It is pointless to return a SingleValidationResult object if there is no error

**Review by fpenim — APPROVED** [review]
👍

**Pull request merged**